### PR TITLE
Add route navigate with matrix params

### PR DESCRIPTION
**Pull request by irustm**

### CODE REVIEW

**Review by fetis — COMMENTED** [review]
~this is explained in docs~

File: snippets/route-navigate-with-matrix-params.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+title: Route navigate with matrix params
```
Comment (fetis) [inline snippets/route-navigate-with-matrix-params.md]: I suggest using word "optional" for these parameters

title might be "Optional parameters in the middle"

File: snippets/route-navigate-with-matrix-params.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+
+# Links
+https://stackblitz.com/edit/angular-xvy5pd
+
+
+
+# ComponentCode
+  import { Component, OnInit } from '@angular/core';
```
Comment (fetis) [inline snippets/route-navigate-with-matrix-params.md]:
```suggestion
import { Component, OnInit } from '@angular/core';
```

File: snippets/route-navigate-with-matrix-params.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    component: FirstComponent,
+    children: [{
+      path: 'details',
+      component: DetailComponent
+    }
+    ]
+  },
+];
+@NgModule({
+  imports:      [ BrowserModule, FormsModule, RouterModule.forRoot(ROUTES) ],
```
Comment (fetis) [inline snippets/route-navigate-with-matrix-params.md]: I would list 1 import/declaration per line

**Review by fetis — CHANGES_REQUESTED** [review]

File: snippets/optional-parameters-in-the-middle.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+title: Optional parameters in the middle
+author: irustm
+level: beginner
+tags:
+  - routing
+---
+# Content
+Navigate with matrix params:
+
+the router will navigate to `/first;name=foo/details`
```
Comment (kirjs) [inline snippets/optional-parameters-in-the-middle.md:11]: What would be the differences/benefits compared to `?name=foo`?

**Review by kirjs — APPROVED** [review]

File: snippets/optional-parameters-in-the-middle.md (hunk 4 above)
Comment (fetis) [reply snippets/optional-parameters-in-the-middle.md:11]: @kirjs as far as I know Angular takes query params as different urls, so when you make link active with `routerLinkActive`  it'll be not highlighted for all cases.

Optional (matrix) params don't have this problem

Comment (irustm) [reply snippets/optional-parameters-in-the-middle.md:11]: You will get matrix params only in the necessary component. And if there is query params, you will get them in all components. Example: https://stackblitz.com/edit/angular-xvy5pd

**Comment by irustm** [comment]
@fetis ?

**Review by fetis — APPROVED** [review]

**Pull request merged**